tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,1639 @@
+LOGGER_CORE = 'core'
+LOGGER_CLI_COMMAND = 'cli_command'
+LOGGER_WEB_PRODUCE = "web_produce"
+LOGGER_WEB_CONSUME = "web_consume"
+
+CONSUMER_LOGGING_THRESHOLD = 100
+
+CONSUME_STOP_EOF = "eof"
+CONSUME_STOP_SLEEP = "sleep"
+CONSUME_STOP_CONTINUOUS = "continuous"
+consume_stop_options = [
+    CONSUME_STOP_EOF, CONSUME_STOP_SLEEP, CONSUME_STOP_CONTINUOUS
+]
+
+
+def setup_logger(child_logger_name: str = "",
+                 level: int = logging.DEBUG) -> logging.Logger:
+    """Create main logger or its child based on naming
+
+    Args:
+        name (str, optional): name of the logger. Defaults to "stream_verifier"
+        level (int, optional): Log level. Defaults to logging.DEBUG.
+
+    Returns:
+        logging.Logger: Logger created
+    """
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setFormatter(
+        logging.Formatter(
+            '%(asctime)s - %(name)s - %(levelname)s - %(message)s'))
+    handler.setLevel(level)
+    logger_name = "stream_verifier"
+    if len(child_logger_name) > 0:
+        # Just simple hierachy, no special handler
+        return logging.getLogger(f"{logger_name}.{child_logger_name}")
+    else:
+        logger = logging.getLogger(logger_name)
+        logger.addHandler(handler)
+        logger.setLevel(level)
+        return logger
+
+
+def write_json(ioclass: IO, data: dict) -> None:
+    ioclass.write(json.dumps(data))
+    ioclass.write('\n')
+    ioclass.flush()
+
+
+def validate_keys(incoming: list[str], local: list[str],
+                  forbidden: list[str]) -> list[str]:
+    error_msgs = []
+    # Validate incoming keys
+    for k in incoming:
+        if k in forbidden:
+            error_msgs += [f"Key '{k}' can't be updated"]
+        elif k not in local:
+            error_msgs += [f"Unknown key '{k}'"]
+    return error_msgs
+
+
+def validate_option(option: str, value: str,
+                    available_values: list[str]) -> Tuple[bool, str]:
+    if value not in available_values:
+        error_msg = f"{option} value of {value} is invalid. " \
+            f"Expected on of: {', '.join(available_values)}"
+        return False, error_msg
+    else:
+        return True, ""
+
+
+class Updateable(object):
+    def update(self, new: dict):
+        for key, value in new.items():
+            if hasattr(self, key):
+                setattr(self, key, value)
+
+
+@dataclass(kw_only=True)
+class AppCfg(Updateable):
+    """Holds configuration for the app along with
+    functions to handle REST calls to update it
+    """
+    app_name: str = app_name
+    brokers: str = "localhost:9092"
+    # topic group id for Consumer configuration
+    topic_group_id: str = "group-stream-verifier-tx"
+    # Topic prefix, full topic name will be
+    # <topic_prefix>-<range(topic_count)>
+    # I.e. "stream-verifier-topic-1", "stream-verifier-topic-2", etc
+    topic_prefix_produce: str = "stream-topic-dst"
+    topic_prefix_consume: str = "stream-topic-src"
+    topic_count: int = 16
```

Comment:
do we need more than one topic? I don't think so for the objective of this test (which is to validate atomicity in the presence of failures), can we remove this configurable? I think that removes a bunch of code around prefixes and such.

Reply:
This is the core of atomic operation. As I understand it we should consume from one topic and produce to the other with deserialization. Not use the same topic for both operations.

Reply:
> As I understand it we should consume from one topic and produce to the other with deserialization.

Right but then we have 16 topics here instead of 1 src and 1 destination?

Reply:
they are working in pairs src-1 -> dst-1, src-2 -> dst-2 and so on

Reply:
I guess my question is we do need so many pairs? Just have 1 source and 1 destination? These multiple source/dst pairs is adding a lot of unnecessary code and will be a nightmare to debug in redpanda logs.

Reply:
This is a part of future high-scale checks, so the goal here is to make it configurable to use either one pair or 100 pairs based on how much pressure we want to have on RP

Reply:
Let's lean on simplicity first, @savex  and build it in a way that we can add high-scale in future. Is that reasonable?

Reply:
Main focus is external simplicity and ease of configurability. It is the same approach with multi-topic approach we used in Many topics testing. And it is already proven useful in finding issues.